source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -263,14 +265,16 @@ std::string SPIFFEValidator::extractTrustDomain(const std::string& san) {
   return "";
 }
 
-size_t SPIFFEValidator::daysUntilFirstCertExpires() const {
+absl::optional<int32_t> SPIFFEValidator::daysUntilFirstCertExpires() const {
   if (ca_certs_.empty()) {
-    return 0;
+    return absl::make_optional(std::numeric_limits<int>::max());
```

Comment:
All these max expressions should use the return value's type, whatever that ends up being, instead of `int`.

Reply:
change to SIZE_MAX.